src/main/java/rlbot/manager/HivemindManager.java

```diff
@@ -0,0 +1,122 @@
+package rlbot.manager;
+
+import rlbot.ControllerState;
+import rlbot.Hivemind;
+import rlbot.cppinterop.RLBotDll;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+/**
+ * This class is a BotManager for hiveminds. It runs the main logic loops and retrieves GameTickPackets for
+ * the hiveminds.
+ */
+public class HivemindManager extends BaseBotManager {
+
+    // There are only every two hiveminds of this bot, once for each team
+    private final HivemindProcess[] hivemindProcesses = new HivemindProcess[2];
```

Comment:
It's theoretically possible that a family (like the Relief family) could have two separate hive implementations (e.g. AdversityHive and AirBudHive) playing on the same team. Possible to support that?

Reply:
First off, AdversityHive sounds scary af.

Secondly, yes-ish. It is currently not supported by the HivemindManager, but the user can still do it right now. They simply have to subdivide their drones based on the names. And they have complete control over how they group the bots, so they could even do pairs of AdversityBots and AirBots. The user has to create that intermediate layer themselves though.

Hm, let me just think about how to support it from the HivemindManager. The python implementation has something called "hivemind keys", and maybe that's a concept that can be reused.

Reply:
There's no point in combining two different types of bot since the will all be part of the hivemind and therefore the only difference would be appearance. (Unless you make the hivemind treat bots differently based on name)

Reply:
The point is, that a family like Relief shares codebase and is thus forced to be the same hive right now. If you want to make multiple hives based on the same codebase, you are in trouble.

Reply:
Yep! This change is awesome though. I don't want to increase the complexity too much with my edge case, so feel free to say "no it's getting too messy," I would still approve.

Reply:
I managed to make it work! I put it on a different branch so we can discuss it separately. See #497.